src/rgw/rgw_data_sync.cc

```diff
@@ -551,15 +551,16 @@ static string full_data_sync_index_shard_oid(const string& source_zone, int shar
 struct bucket_instance_meta_info {
   string key;
   obj_version ver;
-  time_t mtime;
+  utime_t mtime;
   RGWBucketInstanceMetadataObject data;
 
-  bucket_instance_meta_info() : mtime(0) {}
+  bucket_instance_meta_info() {}
 
   void decode_json(JSONObj *obj) {
     JSONDecoder::decode_json("key", key, obj);
     JSONDecoder::decode_json("ver", ver, obj);
-    JSONDecoder::decode_json("mtime", mtime, obj);
+    string s;
+    JSONDecoder::decode_json("mtime", s, obj);
```

Comment:
@cbodley yeah, this is missing the creation of mtime from s. Right, it's encoded in RGWMetadataManager::get().